src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1360,26 +1366,25 @@ private async Task ValidateDatabaseConnection()
                 Connection = conn
             };
 
-            string tablePrefix = GetTablePrefix(conn.Database, schemaName);
-            string queryPrefix = string.IsNullOrEmpty(tablePrefix) ? string.Empty : $"{tablePrefix}.";
+            string tableNameWithPrefix = GetTableNameWithPrefix(schemaName, tableName);
             selectCommand.CommandText
-                = $"SELECT * FROM {queryPrefix}{SqlQueryBuilder.QuoteIdentifier(tableName)}";
+                = $"SELECT * FROM {tableNameWithPrefix}";
             adapterForTable.SelectCommand = selectCommand;
 
-            DataTable[] dataTable = adapterForTable.FillSchema(EntitiesDataSet, SchemaType.Source, tableName);
+            DataTable[] dataTable = adapterForTable.FillSchema(EntitiesDataSet, SchemaType.Source, tableNameWithPrefix);
             return dataTable[0];
         }
 
-        internal string GetTablePrefix(string databaseName, string schemaName)
-        {
+        internal string GetTableNameWithPrefix(string schemaName, string tableName)
```

Comment:
need comment summarizing function + params. also explain what is meant by "prefix"

Reply:
and return value: what is an example of the final string?
- `[schema].[tableName]`
- `'schema'.'tableName'`
- `schema.tableName`